src/Microsoft.NetCore.Analyzers/UnitTests/Security/UseAutoValidateAntiforgeryTokenTests.cs

```diff
@@ -0,0 +1,1033 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    public class UseAutoValidateAntiforgeryTokenTests : DiagnosticAnalyzerTestBase
+    {
+        protected void VerifyCSharpWithDependencies(string source, params DiagnosticResult[] expected)
+        {
+            string microsoftAspNetCoreCSharpSourceCode = @"
```

Comment:
>microsoftAspNetCoreCSharpSourceCode [](start = 19, length = 35)

Put the ASP.NET Core in src\Test.Utilities\MinimalImplementations\, just in case we run across other ASP.NET Core analyzers.